server/aesdsocket.c

```diff
@@ -0,0 +1,363 @@
+    int write_len = 0;
+
+    //Variable to check the file size after writes are performed
+    int file_size = 0;
+
+    struct sockaddr_storage accept_addr;
+
+    socklen_t accept_addr_len = sizeof(accept_addr);
+    
+
+    while(1)
+    {
+        acc_fd = accept(sock_fd, (struct sockaddr *)&accept_addr, &accept_addr_len); //Accept incoming connections in a forever loop as specificed by instructions. 
+
+        if(acc_fd < 0)
+        {
+            perror("accept");
+            continue; //Continue trying to accept if failed
+        }
+
+        //Attempting to print the IP address of the accepted client connection on success of accept using the inet_ntop function
+        struct sockaddr_in *ptr = (struct sockaddr_in *)&accept_addr;
+        syslog(LOG_DEBUG, "Accepted connection from IP : %s", inet_ntop(AF_INET, &ptr->sin_addr, address_str, sizeof(address_str)));
+
+        //Receive all bytes from client
+        while((bytes_received = recv(acc_fd, buffer, sizeof(buffer), 0)) > 0)
+        {
+            printf("Verifying the number of bytes received: %d\n", bytes_received);
+
+            //Check for \n for the end of a packet received. 
+            for(int i = 0; i < bytes_received; i++)
+            {
+                if(buffer[i] == '\n')
+                {
+                    nullterm_flag = 1;
+                    nullchar_it = i+1;
+                    break;
+                }
+            }
+
+            memcpy(packet_buffer + (packet_buffer_size_count -1)*1024, buffer, bytes_received);
+
+            //Check the completeness of data packet and assign bytes to write based on the value of nullchar_it
+            if(nullterm_flag == 1)
+            {
+                write_len = (packet_buffer_size_count -1)*1024 + nullchar_it;
+                break;
+            }
+
+            else
+            {
+                //Increase the size of the packet buffer by 1024 if no Null terminator is found in the buffer to accomodate full packet. 
+                char *tmp = (char *)realloc(packet_buffer, (packet_buffer_len + 1024));
+
+                if(tmp != NULL)
+                {
+                    packet_buffer = tmp;
+                    packet_buffer_len = packet_buffer_len + 1024;
+                    packet_buffer_size_count +=1;
+                }
+
+                else
+                {
+                    perror("Memory allocation failure");
+                }
+            }
+
+        }
+
+        printf("Packet buffer size obtained : %d\n", write_len);
+
+        if(nullterm_flag == 1)
+        {
+            nullterm_flag = 0;
+
+            //Write obtained packet to the file
+            int ret_wr = write(filetotest_fd, packet_buffer, write_len);
+            file_size += ret_wr; //Adding to the current file
+
+            lseek(filetotest_fd, 0, SEEK_SET); //Set the file descriptor to the top of the file
+
+            char *read_buffer = (char*)malloc(file_size);
```

Comment:
@suja8742 what assumption are you making here about the size of the process heap and the total size of all packets sent?